urlparse4/cgurl.pyx

```diff
@@ -177,13 +214,22 @@ def unicode_handling(str):
     return bytes_str
 
 def urlsplit(url):
+    """
+    Use GURL to split the url. Needs to be reviewed!
+    """
+    decode = False if isinstance(url, bytes) else True
     url = unicode_handling(url)
-    return SplitResultNamedTuple.__new__(SplitResultNamedTuple, url)
+    return SplitResultNamedTuple.__new__(SplitResultNamedTuple, url, decode)
 
 def urljoin(base, url, allow_fragments=True):
-    base, url = unicode_handling(base), unicode_handling(url)
-
+    """
+    Use the urljoin function from GURL chromium. Needs to be reviewed!
+    """
+    decode = False if (isinstance(base, bytes) and isinstance(url, bytes)) else True
     if allow_fragments and base:
+        base, url = unicode_handling(base), unicode_handling(url)
+        if decode:
+            return GURL(base).Resolve(url).spec().decode('utf-8')
         return GURL(base).Resolve(url).spec()
-    else:
-        return stdlib_urljoin(base, url, allow_fragments=allow_fragments)
+
+    return stdlib_urljoin(base, url, allow_fragments=allow_fragments)
```

Comment:
Btw, if we have to fallback to stdlib functions sometimes, it makes sense to have the same logic of unicode/bytes handling as in stdlib (which we have now, right?)